src/io/cpal.rs

```diff
@@ -205,14 +236,17 @@ impl AudioBackendManager for CpalBackend {
         }
     }
 
-    fn build_input(options: AudioContextOptions) -> (Self, Receiver<AudioBuffer>)
+    fn build_input(options: AudioContextOptions) -> (Self, Receiver<Vec<f32>>)
```

Comment:
I need to think a bit more about this change, I don't really like flying `Vec` all over the place. I'd rather see we encode the important information alongside it asap (channels, samplerate). The code is a bit messy in `microphone.rs` I notice, I will try if there is another way to make this easier to follow

Reply:
I think the code is a bit more readable like that, but no strong opinion and maybe this is a bit less efficient too. 
I will revert to the previous version, let's keep the mic refactoring for a dedicated issue / PR.